provider/image.go

```diff
@@ -428,6 +438,43 @@ func getCredentials() (map[string]clitypes.AuthConfig, error) {
 	return auths, nil
 }
 
+// Because the authConfigs provided by the host include the `https://` prefix in the map keys, `getRegistryAddrForAuth`
+// ensures we return a registry address that includes the `https://` scheme.
+// While this prefix is not needed for program-provided auth, it is valid regardless, so adding it by default
+// keeps the special case handling to a minimum.
+func getRegistryAddrForAuth(serverName, imgName string) (string, error) {
+	var serverAddr string
+	if serverName == "docker.io" {
+		// if it's dockerhub, we special case it so host config can find the correct registry
+		return "https://index.docker.io/v1/", nil
+	}
+
+	if serverName == "" {
+		// if there is no servername in the registry input, we attempt to build it from the fully qualified image name.
+		addr, _, found := strings.Cut(imgName, "/")
+		if !found {
+			return "", errors.Errorf("image name must be fully qualified: %s", imgName)
+		}
+		if addr == "docker.io" {
+			return "https://index.docker.io/v1/", nil
+		}
+		// we need the full server address for the lookup
+		serverAddr = "https://" + addr
+
+	} else {
+		// check if the provider registry server starts with https://
+		if strings.HasPrefix(serverName, "https://") {
+			serverAddr = serverName
+		} else {
+			// courtesy add the prefix so user does not have to explicitly do so
+			serverAddr = "https://" + serverName
+		}
+		// TODO: this is where we would default to Docker if we wanted to do so.
+		// A valid registry address includes a `.` so we could add additional helper checks here.
```

Comment:
Do we want to default to docker? If we're going to check in a TODO, it would be great to have the TODO associated with an issue.